snakemake/sourcecache.py

```diff
@@ -306,12 +292,7 @@ def infer_source_file(path_or_uri, basedir: SourceFile = None):
             return basedir.join(path_or_uri)
         return LocalSourceFile(path_or_uri)
     if path_or_uri.startswith("git+file:"):
-        try:
-            root_path, file_path, ref = split_git_path(path_or_uri)
-        except Exception as e:
-            raise WorkflowError(
-                f"Failed to read source {path_or_uri} from git repo.", e
-            )
+        root_path, file_path, ref = split_git_path(path_or_uri)
```

Comment:
Why removing the exception handling here?

Reply:
I have readded the check